app/pods/newsletter/index/route.js

```diff
@@ -18,32 +19,51 @@ export default class IndexNewsletterRoute extends Route {
    */
   async model(params) {
     const agenda = this.modelFor('newsletter').agenda; // eslint-disable-line
-    const agendaitems = await this.store
-      .query('agendaitem', {
-        'filter[agenda][:id:]': agenda.id,
-        'filter[show-as-remark]': false,
-        'filter[is-approval]': false,
-        include: 'treatments.newsletter-info',
-        sort: params.sort,
-        'page[size]': PAGE_SIZE.AGENDAITEMS,
-      });
-
-    return Promise.all(agendaitems.map(async(agendaitem) => {
-      const agendaItemTreatments = await agendaitem.get('treatments');
-      const agendaItemTreatment = agendaItemTreatments.firstObject;
-      const newsletterInfo = await agendaItemTreatment.get('newsletterInfo');
-      return {
-        agendaitem,
-        newsletterInfo,
-      };
-    }));
+
+    const filter = {
+      'show-as-remark': false,
+      agenda: {
+        id: agenda.id,
+      },
+      ['is-approval']: false,
+    };
+    let agendaitems = await this.loadAgendaItems(params, filter);
+
+    // KAS-2976 - Old Kaleidos items have undefined is-approval states and need to be queried for this view
+    if (isEmpty(agendaitems)) {
+      filter['is-approval'] = undefined;
+      agendaitems = await this.loadAgendaItems(params, filter);
+    }
+
+    return Promise.all(
+      agendaitems.map(async (agendaitem) => {
+        const agendaItemTreatments = await agendaitem.get('treatments');
+        const agendaItemTreatment = agendaItemTreatments.firstObject;
+        const newsletterInfo = await agendaItemTreatment.get('newsletterInfo');
+        return {
+          agendaitem,
+          newsletterInfo,
+        };
+      })
+    );
+  }
+
+  async afterModel() {
+    this.agenda = this.modelFor('newsletter').agenda;
   }
 
-  async setupController(controller, model) {
+  async setupController(controller) {
     super.setupController(...arguments);
-    const agenda = this.modelFor('newsletter').agenda; // eslint-disable-line
-    controller.set('agenda', agenda);
-    controller.set('model', model);
+    controller.agenda = this.agenda;
+  }
```

Comment:
beetje buiten scope, maar hebben we nood aan die agenda in de controller? lijkt ongebruikt.